tests/trainer/test_trainer_error_handling.py

```diff
@@ -0,0 +1,57 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest import mock
+
+import pytest
+
+from pytorch_lightning import Trainer
+from tests.helpers import BoringModel
+from tests.helpers.runif import RunIf
+
+
+class TrainerStagesErrorsModel(BoringModel):
+    def on_train_start(self) -> None:
+        raise Exception("Error during train")
+
+    def on_validation_start(self) -> None:
+        raise Exception("Error during validation")
+
+    def on_test_start(self) -> None:
+        raise Exception("Error during test")
+
+    def on_predict_start(self) -> None:
+        raise Exception("Error during predict")
+
+
+@pytest.mark.parametrize(
+    "accelerator,num_processes",
+    [
+        (None, 1),
+        pytest.param("ddp_cpu", 2, marks=RunIf(skip_windows=True)),
+        pytest.param("ddp", 2, marks=RunIf(skip_windows=True)),
```

Comment:
won't pass. ddp creates subprocesses (see hanging tests).

for testing with ddp, a separate test would be needed with `RunIf(special=True)`.

Reply:
thanks, I was wondering about that! Fixed!

Reply:
no it did not work because adding RunIf is not enough. I was too brief on explaining this, sorry, but as I said it would really have to be its own separate test. 

a test that runs with ddp spawning multiple subprocesses will not join anymore, meaning the subsequent parameterization will run under the same process group. therefore if one of the parameterizations is a tiny bit faster than the other, it will lead to problems. It could be one of the reasons the test crashed here. 

But the ddp tests are costly and it will cost us several seconds, so I suggest removing the configuration `pytest.param("ddp", 2,)` entirely.

Reply:
will do, thank you for the explanation!